### PR TITLE
DCS-910: email will be sent when involved staff removed

### PR DESCRIPTION
**Pull request by GurnankCheema**
There are two ways an involved staff can be removed from an incident report.
First is by coordinator using the 'delete' link in view-report page and the second
is the coordinator using the 'view removal request' link in view-statements and following the process through.
Ultimately they both do the same thing. The action to send an email to the involved staff member starts in the involvedStaffService.removeInvolvedStaff() method. 
It contains new code notificationService.sendInvolvedStaffRemovedFromReport(...) to send the email.
A new template has been created in Govuk Notify.

example of email
Screenshot 2021-04-21 at 17 38 47

### CODE REVIEW

File: server/config.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ module.exports = {
             REQUEST: 'a58349d2-8a5f-4a22-8cb3-6a1410201a01',
             REMINDER: 'f561f2bc-9a64-4af2-a34c-ebee08c7503b',
             OVERDUE: 'da6a4684-ec87-4ead-8fcd-ec3a126c4926',
+            REMOVED: 'fc24f4e4-a927-492b-9bb4-dfe4f9e6c383',
```
Comment (GurnankCheema) [inline server/config.js:50]: Gov notify key for the REMOVED template

File: server/services/index.ts [hunk 2]
```diff
@@ -40,7 +40,13 @@ const eventPublisher = EventPublisher(telemetryClient)
 const heatmapBuilder = createHeatmapBuilder(prisonClientBuilder)
 const userService = new UserService(prisonClientBuilder, authClientBuilder)
 const notificationService = notificationServiceFactory(eventPublisher)
-const involvedStaffService = new InvolvedStaffService(incidentClient, statementsClient, userService, db.inTransaction)
+const involvedStaffService = new InvolvedStaffService(
+  incidentClient,
+  statementsClient,
+  userService,
+  db.inTransaction,
+  notificationService
```
Comment (GurnankCheema) [inline server/services/index.ts:48]: Injecting notificationService

File: server/services/involvedStaffService.ts [hunk 3]
```diff
@@ -105,5 +108,15 @@ export class InvolvedStaffService {
         }
       }
     })
+
+    const { submittedDate, email, name: involvedName } = involvedStaff
```
Comment (GurnankCheema) [inline server/services/involvedStaffService.ts:112]: Use the statementId to get get the statement and its parent report

File: server/services/involvedStaffService.ts [hunk 4]
```diff
@@ -105,5 +108,15 @@ export class InvolvedStaffService {
...
+    const incidentDate = moment(involvedStaff.incidentDate).toDate()
+    const context = { reportId, statementId }
+
+    await this.notificationService.sendInvolvedStaffRemovedFromReport(
```
Comment (GurnankCheema) [inline server/services/involvedStaffService.ts:116]: Pass data from above to the template in Notify

File: server/services/reportDetailBuilder.test.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ import ReportDetailBuilder from './reportDetailBuilder'
 
 jest.mock('.')
 
-const involvedStaffService = new InvolvedStaffService(null, null, null, null) as jest.Mocked<InvolvedStaffService>
+const involvedStaffService = new InvolvedStaffService(null, null, null, null, null) as jest.Mocked<InvolvedStaffService>
```
Comment (GurnankCheema) [inline server/services/reportDetailBuilder.test.ts:8]: fix breaking tests

File: server/services/involvedStaffService.ts [hunk 6]
```diff
@@ -88,6 +89,8 @@ export class InvolvedStaffService {
   public async removeInvolvedStaff(reportId: number, statementId: number): Promise<void> {
     logger.info(`Removing statement: ${statementId} from report: ${reportId}`)
 
+    const involvedStaff = await this.statementsClient.getInvolvedStaffToRemove(statementId)
+
     await this.inTransaction(async client => {
       const pendingStatementBeforeDeletion = await this.statementsClient.getNumberOfPendingStatements(reportId, client)
```
Comment (bell-pa) [inline server/services/involvedStaffService.ts:95]: Don't think this line is needed.

**Review by bell-pa — COMMENTED** [review]
LGTM

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**